// @vertx/web/index.js
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/web" />
module.exports = {
  AuthHandler: Java.type('io.vertx.ext.web.handler.AuthHandler'),
  BasicAuthHandler: Java.type('io.vertx.ext.web.handler.BasicAuthHandler'),
  BodyHandler: Java.type('io.vertx.ext.web.handler.BodyHandler'),
  BridgeEvent: Java.type('io.vertx.ext.web.handler.sockjs.BridgeEvent'),
  CSRFHandler: Java.type('io.vertx.ext.web.handler.CSRFHandler'),
  ChainAuthHandler: Java.type('io.vertx.ext.web.handler.ChainAuthHandler'),
  ClusteredSessionStore: Java.type('io.vertx.ext.web.sstore.ClusteredSessionStore'),
  Cookie: Java.type('io.vertx.ext.web.Cookie'),
  CookieHandler: Java.type('io.vertx.ext.web.handler.CookieHandler'),
  CorsHandler: Java.type('io.vertx.ext.web.handler.CorsHandler'),
  DigestAuthHandler: Java.type('io.vertx.ext.web.handler.DigestAuthHandler'),
  ErrorHandler: Java.type('io.vertx.ext.web.handler.ErrorHandler'),
  FaviconHandler: Java.type('io.vertx.ext.web.handler.FaviconHandler'),
  FileUpload: Java.type('io.vertx.ext.web.FileUpload'),
  FormLoginHandler: Java.type('io.vertx.ext.web.handler.FormLoginHandler'),
  JWTAuthHandler: Java.type('io.vertx.ext.web.handler.JWTAuthHandler'),
  LanguageHeader: Java.type('io.vertx.ext.web.LanguageHeader'),
  LocalSessionStore: Java.type('io.vertx.ext.web.sstore.LocalSessionStore'),
  Locale: Java.type('io.vertx.ext.web.Locale'),
  LoggerHandler: Java.type('io.vertx.ext.web.handler.LoggerHandler'),
  MIMEHeader: Java.type('io.vertx.ext.web.MIMEHeader'),
  MultiTenantHandler: Java.type('io.vertx.ext.web.handler.MultiTenantHandler'),
  OAuth2AuthHandler: Java.type('io.vertx.ext.web.handler.OAuth2AuthHandler'),
  ParsedHeaderValue: Java.type('io.vertx.ext.web.ParsedHeaderValue'),
  ParsedHeaderValues: Java.type('io.vertx.ext.web.ParsedHeaderValues'),
  RedirectAuthHandler: Java.type('io.vertx.ext.web.handler.RedirectAuthHandler'),
  ResponseContentTypeHandler: Java.type('io.vertx.ext.web.handler.ResponseContentTypeHandler'),
  ResponseTimeHandler: Java.type('io.vertx.ext.web.handler.ResponseTimeHandler'),
  Route: Java.type('io.vertx.ext.web.Route'),
  Router: Java.type('io.vertx.ext.web.Router'),
  RoutingContext: Java.type('io.vertx.ext.web.RoutingContext'),
  Session: Java.type('io.vertx.ext.web.Session'),
  SessionHandler: Java.type('io.vertx.ext.web.handler.SessionHandler'),
  SessionStore: Java.type('io.vertx.ext.web.sstore.SessionStore'),
  SockJSHandler: Java.type('io.vertx.ext.web.handler.sockjs.SockJSHandler'),
  SockJSSocket: Java.type('io.vertx.ext.web.handler.sockjs.SockJSSocket'),
  StaticHandler: Java.type('io.vertx.ext.web.handler.StaticHandler'),
  TemplateEngine: Java.type('io.vertx.ext.web.templ.TemplateEngine'),
  TemplateHandler: Java.type('io.vertx.ext.web.handler.TemplateHandler'),
  TimeoutHandler: Java.type('io.vertx.ext.web.handler.TimeoutHandler'),
  UserSessionHandler: Java.type('io.vertx.ext.web.handler.UserSessionHandler'),
  VirtualHostHandler: Java.type('io.vertx.ext.web.handler.VirtualHostHandler')
};
