// @vertx/web/module.mjs
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/web" />
export const AuthHandler = Java.type('io.vertx.ext.web.handler.AuthHandler');
export const BasicAuthHandler = Java.type('io.vertx.ext.web.handler.BasicAuthHandler');
export const BodyHandler = Java.type('io.vertx.ext.web.handler.BodyHandler');
export const BridgeEvent = Java.type('io.vertx.ext.web.handler.sockjs.BridgeEvent');
export const CSRFHandler = Java.type('io.vertx.ext.web.handler.CSRFHandler');
export const ChainAuthHandler = Java.type('io.vertx.ext.web.handler.ChainAuthHandler');
export const ClusteredSessionStore = Java.type('io.vertx.ext.web.sstore.ClusteredSessionStore');
export const Cookie = Java.type('io.vertx.ext.web.Cookie');
export const CookieHandler = Java.type('io.vertx.ext.web.handler.CookieHandler');
export const CorsHandler = Java.type('io.vertx.ext.web.handler.CorsHandler');
export const DigestAuthHandler = Java.type('io.vertx.ext.web.handler.DigestAuthHandler');
export const ErrorHandler = Java.type('io.vertx.ext.web.handler.ErrorHandler');
export const FaviconHandler = Java.type('io.vertx.ext.web.handler.FaviconHandler');
export const FileUpload = Java.type('io.vertx.ext.web.FileUpload');
export const FormLoginHandler = Java.type('io.vertx.ext.web.handler.FormLoginHandler');
export const JWTAuthHandler = Java.type('io.vertx.ext.web.handler.JWTAuthHandler');
export const LanguageHeader = Java.type('io.vertx.ext.web.LanguageHeader');
export const LocalSessionStore = Java.type('io.vertx.ext.web.sstore.LocalSessionStore');
export const Locale = Java.type('io.vertx.ext.web.Locale');
export const LoggerHandler = Java.type('io.vertx.ext.web.handler.LoggerHandler');
export const MIMEHeader = Java.type('io.vertx.ext.web.MIMEHeader');
export const MultiTenantHandler = Java.type('io.vertx.ext.web.handler.MultiTenantHandler');
export const OAuth2AuthHandler = Java.type('io.vertx.ext.web.handler.OAuth2AuthHandler');
export const ParsedHeaderValue = Java.type('io.vertx.ext.web.ParsedHeaderValue');
export const ParsedHeaderValues = Java.type('io.vertx.ext.web.ParsedHeaderValues');
export const RedirectAuthHandler = Java.type('io.vertx.ext.web.handler.RedirectAuthHandler');
export const ResponseContentTypeHandler = Java.type('io.vertx.ext.web.handler.ResponseContentTypeHandler');
export const ResponseTimeHandler = Java.type('io.vertx.ext.web.handler.ResponseTimeHandler');
export const Route = Java.type('io.vertx.ext.web.Route');
export const Router = Java.type('io.vertx.ext.web.Router');
export const RoutingContext = Java.type('io.vertx.ext.web.RoutingContext');
export const Session = Java.type('io.vertx.ext.web.Session');
export const SessionHandler = Java.type('io.vertx.ext.web.handler.SessionHandler');
export const SessionStore = Java.type('io.vertx.ext.web.sstore.SessionStore');
export const SockJSHandler = Java.type('io.vertx.ext.web.handler.sockjs.SockJSHandler');
export const SockJSSocket = Java.type('io.vertx.ext.web.handler.sockjs.SockJSSocket');
export const StaticHandler = Java.type('io.vertx.ext.web.handler.StaticHandler');
export const TemplateEngine = Java.type('io.vertx.ext.web.templ.TemplateEngine');
export const TemplateHandler = Java.type('io.vertx.ext.web.handler.TemplateHandler');
export const TimeoutHandler = Java.type('io.vertx.ext.web.handler.TimeoutHandler');
export const UserSessionHandler = Java.type('io.vertx.ext.web.handler.UserSessionHandler');
export const VirtualHostHandler = Java.type('io.vertx.ext.web.handler.VirtualHostHandler');
