// @vertx/unit/module.mjs
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/unit" />
export const Async = Java.type('io.vertx.ext.unit.Async');
export const Completion = Java.type('io.vertx.ext.unit.Completion');
export const EventBusCollector = Java.type('io.vertx.ext.unit.collect.EventBusCollector');
export const Failure = Java.type('io.vertx.ext.unit.report.Failure');
export const TestCase = Java.type('io.vertx.ext.unit.TestCase');
export const TestCaseReport = Java.type('io.vertx.ext.unit.report.TestCaseReport');
export const TestCompletion = Java.type('io.vertx.ext.unit.TestCompletion');
export const TestContext = Java.type('io.vertx.ext.unit.TestContext');
export const TestResult = Java.type('io.vertx.ext.unit.report.TestResult');
export const TestSuite = Java.type('io.vertx.ext.unit.TestSuite');
export const TestSuiteReport = Java.type('io.vertx.ext.unit.report.TestSuiteReport');
